cli/wallet/wallet.go

```diff
@@ -542,27 +557,61 @@ func importMultisig(ctx *cli.Context) error {
 		}
 	}
 
-	var label *string
 	if ctx.IsSet("name") {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
+	for _, pub := range pubs {
+		match := findAccountsByPublicKey(wall, pub)
+		if len(match) > 0 {
+			matchingAccounts = append(matchingAccounts, match...)
+			accEncPub = pub
+		}
 	}
 
-	if err := addAccountAndSave(wall, acc); err != nil {
-		return cli.NewExitError(err, 1)
+	if len(matchingAccounts) == 1 {
+		acc = matchingAccounts[0]
+		err = acc.ConvertMultisigEncrypted(accEncPub, m, pubs)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		wall.AddAccount(acc)
+		if err := wall.Save(); err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		return nil
 	}
+	if ctx.IsSet("wif") {
+		acc, err = newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		if err := acc.ConvertMultisig(m, pubs); err != nil {
+			return cli.NewExitError(err, 1)
+		}
 
+		if err := addAccountAndSave(wall, acc); err != nil {
+			return cli.NewExitError(err, 1)
+		}
+	} else {
+		return cli.NewExitError(errors.New("none of the provided public keys correspond to an existing key in the wallet, and no WIF is provided"), 1)
+	}
 	return nil
 }
 
+// findAccountsByPublicKey returns all accounts in the given wallet that correspond
+// to the given public key.
+func findAccountsByPublicKey(wall *wallet.Wallet, pub *keys.PublicKey) []*wallet.Account {
+	var accounts = make([]*wallet.Account, 0, 1)
+	for _, acc := range wall.Accounts {
+		if acc.ScriptHash().Equals(pub.GetScriptHash()) {
+			accounts = append(accounts, acc)
+		}
+	}
+	return accounts
+}
```

Comment:
It's so small that I think we don't need a separate helper for it. Move this code to `importMultisig`, it's just two nested `for` loops.